### PR TITLE
Fix #12365: Company Window now displays proper inauguration year and period while in wallclock mode

### LINKED ISSUE #12365: [Bug]: Company inauguration date should use calendar year in wallclock mode
### Version of OpenTTD

14.0-RC3

### Expected result

Company inauguration date, similar to eg. vehicle design dates, should probably always be displayed as calendar year.

### Actual result

image

In wallclock mode, inauguration time is in minutes(?) with no units, and looks quite weird.

### Steps to reproduce

1. Set the time setting to wallclock mode
2. Start a new game and look at the company information window

### PR DESCRIPTION
**Pull request by Vinixss**
## Motivation / Problem
When using the Wallclock timekeeping setting, the Inauguration year of the company is displayed in minutes rather than the actual calendar year.

Closes https://github.com/OpenTTD/OpenTTD/issues/12365.

## Description
Added a new variable in company_base.h named inaugurated_year_calendar to store the calendar version of the company's inaugurated year. If the player is in wallclock mode, this the value of inaugurated_year_calendar and inaugurated_year are passed into a widget to be displayed within the Company window.

## Limitations
Problem appears to be solved in all scenarios and is complete.

## Checklist for review
This PR touches english.txt

### CODE REVIEW

**Comment by Vinixss** [comment]
I closed the former pull request and created a new fork as I messed up too much on the commits to the previous fork, so I believed it to be bet to start from scratch. Sorry for the inconvenience.

File: src/saveload/afterload.cpp [hunk 1]
```diff
@@ -3263,6 +3263,12 @@ bool AfterLoadGame()
 		ScriptObject::InitializeRandomizers();
 	}
 
+	if (IsSavegameVersionBefore(SLV_COMPANY_INAUGURATED_PERIOD)) {
+		for (Company *c : Company::Iterate()) {
+			c->inaugurated_year_calendar = 1950;
```
Comment (2TallTyler) [inline src/saveload/afterload.cpp]: We do have a nice constant for the original start year of 1920: `CalendarTime::ORIGINAL_BASE_YEAR`.

That said, I'd suggest setting it to the starting year of this game, which is generally accurate for single-player games (the most likely type of older save to get loaded). 🙂 

```suggestion
			c->inaugurated_year_calendar = _settings_game.game_creation.starting_year;
```

**Review by 2TallTyler — DISMISSED** [review]
Code looks good and works in my tests. Thank you for the contribution!

(Next step is to wait for another core developer to review and merge. There is no need to update anything until this happens, unless there's a merge conflict -- if that happens we'll explain what to do. 😄)

File: src/company_gui.cpp [hunk 2]
```diff
@@ -2401,7 +2401,14 @@ struct CompanyWindow : Window
 				break;
 
 			case WID_C_DESC_INAUGURATION:
-				SetDParam(0, Company::Get((CompanyID)this->window_number)->inaugurated_year);
+				if (TimerGameEconomy::UsingWallclockUnits()) {
+					SetDParam(0, STR_COMPANY_VIEW_INAUGURATED_TITLE_WALLCLOCK);
+					SetDParam(1, Company::Get((CompanyID) this->window_number)->inaugurated_year_calendar);
+					SetDParam(2, Company::Get((CompanyID) this->window_number)->inaugurated_year);
```
Comment (rubidium42) [inline src/company_gui.cpp]: I know the old version used C-style casts, but we try to slowly get rid of them. So lets not add any new ones.
```suggestion
					SetDParam(1, Company::Get(static_cast<CompanyID>(this->window_number))->inaugurated_year_calendar);
					SetDParam(2, Company::Get(static_cast<CompanyID>(this->window_number))->inaugurated_year);
```

File: src/company_gui.cpp [hunk 3]
```diff
@@ -2401,7 +2401,14 @@ struct CompanyWindow : Window
...
+				} else {
+					SetDParam(0, STR_COMPANY_VIEW_INAUGURATED_TITLE);
+					SetDParam(1, Company::Get((CompanyID) this->window_number)->inaugurated_year);
```
Comment (rubidium42) [inline src/company_gui.cpp]:
```suggestion
					SetDParam(1, Company::Get(static_cast<CompanyID>(this->window_number))->inaugurated_year);
```

**Review by rubidium42 — COMMENTED** [review]
Also requires a rebase since a new saveload version has been introduced.

**Comment by Vinixss** [comment]
Sorry that took so long. I'm not sure if I need to do anything else with saveload or if I have to rebase some commits, so just let me know. It won't take 2 to 3 weeks this time, I swear.

**Comment by 2TallTyler** [comment]
You need to rebase to resolve the merge conflict.

Assuming you've set up your remotes as described in [our contributors' guide](https://github.com/OpenTTD/OpenTTD/blob/master/CONTRIBUTING.md), you'll need to run `git fetch upstream`, then `git rebase upstream/master`. This will present a conflict in `saveload.h` for you to resolve before running `git rebase --continue` to finish the rebase.

You will then need to squash the commit you previously added, so this PR is just one commit again.

**Comment by Vinixss** [comment]
Believe I've got it right now. Sorry for the trouble

File: src/saveload/saveload.h [hunk 4]
```diff
@@ -381,9 +381,13 @@ enum SaveLoadVersion : uint16_t {
 
 	SLV_COMPANY_ALLOW_LIST,                 ///< 335  PR#12337 Saving of list of client keys that are allowed to join this company.
 	SLV_GROUP_NUMBERS,                      ///< 336  PR#12297 Add per-company group numbers.
+<<<<<<< HEAD
 	SLV_INCREASE_STATION_TYPE_FIELD_SIZE,   ///< 337  PR#12572 Increase size of StationType field in map array
 	SLV_ROAD_WAYPOINTS,                     ///< 338  PR#12572 Road waypoints
 
+	SLV_COMPANY_INAUGURATED_PERIOD,         ///< 339  PR#12798 Companies show the period inaugurated in wallclock mode.
+>>>>>>> 4a745b8737 (Codechage #12365|2402114: Moved Inaugurated Period saveload and changed comment to reflect the current master branch)
```
Comment (2TallTyler) [inline src/saveload/saveload.h]: This conflict wasn't quite resolved. It should look like this:

```suggestion
	SLV_INCREASE_STATION_TYPE_FIELD_SIZE,   ///< 337  PR#12572 Increase size of StationType field in map array
	SLV_ROAD_WAYPOINTS,                     ///< 338  PR#12572 Road waypoints
	SLV_COMPANY_INAUGURATED_PERIOD,         ///< 339  PR#12798 Companies show the period inaugurated in wallclock mode.
```

**Review by 2TallTyler — COMMENTED** [review]
Please make sure your commit name is not enclosed in quotation marks: that won't pass the commit checker.

You're so close! One more push! 😄

File: src/saveload/saveload.h [hunk 5]
```diff
@@ -381,8 +381,10 @@ enum SaveLoadVersion : uint16_t {
 
 	SLV_COMPANY_ALLOW_LIST,                 ///< 335  PR#12337 Saving of list of client keys that are allowed to join this company.
 	SLV_GROUP_NUMBERS,                      ///< 336  PR#12297 Add per-company group numbers.
+
```
Comment (2TallTyler) [inline src/saveload/saveload.h]: Extra newline snuck in here.

```suggestion
```

**Review by 2TallTyler — COMMENTED** [review]
Commit name is correct, code looks good (and works), remove that pesky newline and it's good to go! 😄

File: src/saveload/afterload.cpp [hunk 6]
```diff
@@ -3272,6 +3272,12 @@ bool AfterLoadGame()
 		ScriptObject::InitializeRandomizers();
 	}
 
+	if (IsSavegameVersionBefore(SLV_COMPANY_INAUGURATED_PERIOD)) {
+		for (Company *c : Company::Iterate()) {
+			c->inaugurated_year_calendar.c = _settings_game.game_creation.starting_year;
```
Comment (2TallTyler) [inline src/saveload/afterload.cpp]: Heheh, I missed this typo until the build actions failed. 😄 

```suggestion
			c->inaugurated_year_calendar = _settings_game.game_creation.starting_year;
```

**Comment by Vinixss** [comment]
My bad. I should have built everything to make sure it all worked

**Review by 2TallTyler — APPROVED** [review]
Thanks for your hard work getting through the review process!

From here, just wait for another developer to review and merge this PR (or suggest changes that I missed). You don't need to click any more buttons unless we ask you to. (Explaining this because sometimes people try to keep it updated and make a mess of Git. 😄)

**Pull request merged**